src/testdrive/src/action/sql_server/connect.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_trait::async_trait;
+
+use tiberius::{Client, Config};
+use tokio::net::TcpStream;
+use tokio_util::compat::TokioAsyncWriteCompatExt;
+
+use crate::action::{Action, State};
+use crate::parser::BuiltinCommand;
+
+pub struct ConnectAction {
+    name: String,
+    config: tiberius::Config,
+}
+
+pub fn build_connect(mut cmd: BuiltinCommand) -> Result<ConnectAction, String> {
+    let name = cmd.args.string("name")?;
+    cmd.args.done()?;
+
+    let ado_string = cmd.input.join("\n");
```

Comment:
The ADO string can contain spaces, so it needs to be fetched from the body of the command, not from an argument.